iOSDFULibrary/Classes/Utilities/Streams/DFUStreamHex.swift

```diff
@@ -49,15 +49,24 @@ internal class DFUStreamHex : DFUStream {
     }
     
     init(urlToHexFile: URL, urlToDatFile: URL?, type: DFUFirmwareType) {
-        let hexData = try? Data(contentsOf: urlToHexFile)
+        let hexData = try! Data(contentsOf: urlToHexFile)
```

Comment:
What if `urlToHexFile` is empty or not valid ? try will throw an exception that's not handled here

Reply:
It would have crashed on the next lie either way. It's up to the user to ensure that the URL is OK.

Reply:
Ok, so rethrowing the exception is not a choice ? I was thinking of marking the init method as throws, then we can handle that instead of causing a crash. might be marked for a later task as an enhacement, but i get your point that you should be able to select the file and send the proper URL in the first place